### DISCUSSION TITLE
"activate script" checkbox is unchecked after reboot

### DISCUSSION THREAD

**Discussion by Kugeleis** [arendst/Tasmota#16151] [Q&A] [answered]
For whatever reason my nodemcu reboots occasionally. After that the "activate script" is unchecked and my script is not startting. Is this a feature or a bug? How can I ensure my script starts after a reboot?
(Tasmota Version 12.0.2(TasmoCompiler-esp32generic) )

**Comment by sfromis** [comment, marked as answer]
Check the logs while booting.

A common recovery mechanism is that in case of a boot loop (with Tasmota repeatedly crashing shortly after boot), features will gradually be disabled till able to boot without a crash, with messages like `FRC: Some settings have been reset (4)`; the number indicating number of retries.

Better than staying in the boot loop "forever". One potential cause, outside of software-related crashes, is a dubious power situation.

**Comment by Kugeleis** [comment]
OK, I installed rsyslog on my little server and start listening to tasmota.

**Reply by sfromis** [reply]
Boot loop crashes will generally occur before connecting to Wifi, meaning that a syslog server may not see much.

**Reply by Kugeleis** [reply]
So what would be an option except staring at the screen? The device is at the smart meter hidden in a box.

**Reply by sfromis** [reply]
In that case, you have no screen to stare at. Web console suffers from the same issue with normally only coming online after successful boot. You could check if RestartReason says anything, but that generally works best when there is no boot loop, and hence no recovery-reset of config.

What works best is to have a serial terminal connected, and checking the messages after having had the issue.

**Reply by joba-1** [reply]
Do you have another esp around?
Then you could try https://tasmota.github.io/docs/Serial-to-TCP-Bridge/

Edit: a power issue is likely, so best power the bridge esp independent from the device under test (battery?).
